tests/sources/test_dropbox_aiohttp.py

```diff
@@ -0,0 +1,809 @@
+    ):
+        with pytest.raises(ClientResponseError):
+            await anext(
+                source.dropbox_client.api_call(
+                    base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                    url_name=PING,
+                    data=json.dumps(None),
+                )
+            )
+        await source.close()
+
+
+@pytest.mark.asyncio
+async def test_api_call_when_status_429_exception():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client.retry_count = 0
+
+    source.dropbox_client._set_access_token = AsyncMock()
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        side_effect=ClientResponseError(
+            status=429,
+            headers={"Retry-After": 0},
+            request_info=aiohttp.RequestInfo(
+                real_url="", method=None, headers=None, url=""
+            ),
+            history=(),
+        ),
+    ):
+        _ = source.dropbox_client._get_session
+        with pytest.raises(ClientResponseError):
+            await anext(
+                source.dropbox_client.api_call(
+                    base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                    url_name=PING,
+                    data=json.dumps(None),
+                )
+            )
+        await source.close()
+
+
+@pytest.mark.asyncio
+@patch("connectors.sources.dropbox_aiohttp.DEFAULT_RETRY_AFTER", 0)
+async def test_api_call_when_status_429_exception_without_retry_after_header():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client.retry_count = 0
+
+    source.dropbox_client._set_access_token = AsyncMock()
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        side_effect=ClientResponseError(
+            status=429,
+            headers={},
+            request_info=aiohttp.RequestInfo(
+                real_url="", method=None, headers=None, url=""
+            ),
+            history=(),
+        ),
+    ):
+        _ = source.dropbox_client._get_session
+        with pytest.raises(ClientResponseError):
+            await anext(
+                source.dropbox_client.api_call(
+                    base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                    url_name=PING,
+                    data=json.dumps(None),
+                )
+            )
+        await source.close()
+
+
+@pytest.mark.asyncio
+async def test_get_content_when_is_downloadable_is_true():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client._set_access_token = AsyncMock()
+
+    with mock.patch("aiohttp.ClientSession.post", return_value=get_stream_reader()):
+        with mock.patch(
+            "aiohttp.StreamReader.iter_chunked",
+            return_value=AsyncIterator([bytes(RESPONSE_CONTENT, "utf-8")]),
+        ):
+            response = await source.get_content(
+                attachment=MOCK_ATTACHMENT,
+                doit=True,
+            )
+            assert response == EXPECTED_CONTENT
```

Comment:
All these content tests repeat a lot of code, and could be combined into one test with a `@pytest.mark.parametrize`